lib/collection/variable-scope.js

```diff
@@ -126,13 +134,26 @@ _.assign(VariableScope.prototype, /** @lends VariableScope.prototype */ {
     },
 
     /**
-     * Fetches a variable from the current scope.
+     * Fetches a variable from the current scope or from parent scopes if present.
      *
-     * @param {String} key - The name of the variable to set.
-     * @returns {*} The value of the specified variable in the current context.
+     * @param {String} key - The name of the variable to get.
+     * @returns {*} The value of the specified variable across scopes.
      */
     get: function (key) {
         var variable = this.values.one(key);
+
+        // if a variable does not exist in local scope, we search all the layers and find the first occurence.
+        if (!variable) {
+            _.forEach(this.layers, function (layer) {
```

Comment:
Better use a `for` loop here, this is a low-level function, so we don't need the additional overhead of multiple function calls.

Also, then you can simply return from the loop itself, making the code simpler.